packages/engine/Source/DataSources/FeatureProvider.js

```diff
@@ -0,0 +1,42 @@
+import defined from "../Core/defined.js";
+import FeatureLayer from "./FeatureLayer.js";
+
+/**
+ * A class for things
+ * @constructor
+ * @param {Object} options
+ */
+function FeatureProvider(options) {
+  this.id = options.id;
```

Comment:
We've had some pain points with inheritance in the past. To avoid these, I would recommend treating `FeatureProvider` as an interface as much as possible. That is, I would avoid defining _any_ implementation details, and leave it to the implementor or static utility-like functions.